common/djangoapps/entitlements/models.py

```diff
@@ -302,6 +306,76 @@ def get_active_entitlements_for_user(cls, user):
             enrollment_course_run=None
         ).select_related('user').select_related('enrollment_course_run')
 
+    @classmethod
+    def does_user_have_active_entitlements(cls, user):
+        """
+        Checks whether or not a user has any active entitlements
+        """
+        count = cls.objects.count(
+            user=user
+        ).exclude(
+            expired_at__isnull=False,
+            enrollment_course_run=None
+        ).select_related('user').select_related('enrollment_course_run')
+        return count > 0
+
+    @classmethod
+    def get_fullfillable_entitlement_for_user(cls, user, course_run_key):
+        """
+        Looks at the User's existing entitlements to see if the user already has a Course Entitlement for the
+        course run provided in the course_key.  If the user does have an Entitlement with no run set, the User is
+        enrolled in the mode set in the Entitlement.
+
+        Note: This method assumes that eligibility to enroll in the course has already been checked.
+
+        Arguments:
+            user (User): The user that we are inspecting the entitlements for.
+            course_run_key (CourseKey): The course run Key.
+
+        Returns:
+            CourseEntitlement: The most recent fullfillale CourseEntitlement, None otherwise.
+        """
+        # Check if the User has any Entitlements that are active
+        if cls.does_user_have_active_entitlements(user):
```

Comment:
Seems like this conditional might be redundant, could it be removed? right below this check you have a filter that does almost the same thing but slightly more specific.

Reply:
This check was meant to say if the user does not have entitlements we should not continue. Mainly to avoid the API calls to the discovery service if we do not need them.

Reply:
Also the filtered retrieval below applies a more complex filter.  The main goal was to avoid API calls if we do not need them.